Tests iOS/Support/SnowplowMicro.swift

```diff
@@ -0,0 +1,322 @@
+
+    /**
+     Finds the first context of the specified schema type. There is only ever 1 of each type on any given event.
+     */
+    func getContext(of type: String) -> SnowplowMicroContext? {
+        return getContexts().first(where: { $0.schema == type })
+    }
+
+    /**
+     Pulls the API User out of the event
+     */
+    func getAPIUserContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/api_user/jsonschema/1-0-1")
+    }
+
+    /**
+     Pulls the user out of the event
+     */
+    func getUserContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/user/jsonschema/1-0-0")
+    }
+
+    /**
+     Pulls the UI out of the event
+     */
+    func getUIContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/ui/jsonschema/1-0-3")
+    }
+
+    /**
+     Pulls the content out of the event
+     */
+    func getContentContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/content/jsonschema/1-0-0")
+    }
+
+    /**
+     Pulls the slate out of the event
+     */
+    func getSlateContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/slate/jsonschema/1-0-0")
+    }
+
+    /**
+     Pulls the slate lineup out of the event
+     */
+    func getSlateLineupContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/slate_lineup/jsonschema/1-0-0")
+    }
+
+    /**
+     Pulls the recommendation out of the event
+     */
+    func getRecommendationContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/recommendation/jsonschema/1-0-0")
+    }
+
+    /**
+     Pulls the report out of the event
+     */
+    func getReportContext() -> SnowplowMicroContext? {
+        return getContext(of: "iglu:com.pocket/report/jsonschema/1-0-0")
+    }
+}
+
+struct SnowplowMicroEventData: Codable {
+    var unstruct_event: SnowplowMicroUnstructEvent
+    var contexts: SnowplowMicroEventContext
+}
+
+struct SnowplowMicroEventContext: Codable {
+    var schema: String
+    var data: [SnowplowMicroContext]
+}
+
+struct SnowplowMicroUnstructEvent: Codable {
+    var schema: String
+    var data: SnowplowMicroContext
+}
+
+/**
+ The snowplow struct that contains the data we would validate under data dict
+ In the future we could use schema value and Codable to serialize these back into the objects we use in Snowplow for easier validation.
+ */
+struct SnowplowMicroContext: Codable {
+    var schema: String
+    var data: AnyCodable
+
+    func dataDict() -> [String: Any?] {
+        return data.value as! [String: Any?]
+    }
+
+    func has(identifier: String) -> Bool {
+        return (self.dataDict()["identifier"] as? String) == identifier
+    }
+
+    func has(url: String) -> Bool {
+        return (self.dataDict()["url"] as? String) == url
+    }
+
+    func assertHas(url: String) {
+        return XCTAssertEqual((self.dataDict()["url"] as? String), url)
```

Comment:
Doing the assert here, so that when it fails we get the inequality printed instead of using `has` and doing an isTrue Assertion